src/app/barter/components/user-transactions/user-transactions.component.ts

```diff
@@ -0,0 +1,185 @@
+              private modalService: MDBModalService,
+              private transactionStore$: Store<TransactionsState>,
+              private itemsStore$: Store<ItemsState>,
+              private userStore$: Store<UserState>,
+              private alertsService: AlertsService) {
+  }
+
+  ngOnInit() {
+    this.activatedRoute.data.subscribe((result) => {
+      result.data.forEach(transaction =>
+        this.elements.push({
+          id: transaction.id,
+          bidder: transaction.trader,
+          owner: transaction.owner,
+          offeredItem: transaction.traderItem,
+          ownerItem: transaction.ownerItem,
+          isCompleted: transaction.isCompleted,
+          inProgress: transaction.inProgress
+        }));
+
+      this.elements.sort((a, b) => a.isCompleted - b.isCompleted);
+
+      this.transactionsElements = this.elements;
+      this.setTableData();
+    });
+
+    this.userStore$.select(getUser).subscribe((user: User) => this.currentUser = user);
+  }
+
+  ngAfterViewInit() {
+    this.mdbTablePagination.setMaxVisibleItemsNumberTo(this.maxTableRows);
+    this.mdbTablePagination.calculateFirstItemIndex();
+    this.mdbTablePagination.calculateLastItemIndex();
+    this.cdRef.detectChanges();
+  }
+
+  showItemDetails(item: Item): void {
+    this.modalRef = this.modalService.show(ItemDetailsModalComponent, {
+      data: {
+        item,
+        isWatchMode: true
+      },
+      class: 'modal-lg'
+    });
+  }
+
+  acceptOffer(transaction): void {
+    this.itemsStore$.select(getTransactionItems(transaction.ownerItem.id, transaction.offeredItem.id)).pipe(first())
+      .subscribe((items: Item[]) => {
+        const firstItem: Item = {...items[0]};
+        const secondItem: Item = {...items[1]};
+
+        const firstItemOwnerId = firstItem.ownerId;
+        firstItem.ownerId = secondItem.ownerId;
+        secondItem.ownerId = firstItemOwnerId;
+
+        const transactionIndex = this.elements.indexOf(this.elements.find(element =>
+          (element.offeredItem.id === firstItem.id || element.offeredItem.id === secondItem.id) &&
+          ((element.ownerItem.id === firstItem.id || element.ownerItem.id === secondItem.id))));
+        this.elements[transactionIndex].isCompleted = true;
+
+        this.itemsStore$.dispatch(updateItem({item: firstItem}));
+        this.itemsStore$.dispatch(updateItem({item: secondItem}));
+        this.updateTransactionStatus(transaction.id);
+        this.declineOtherTransactions(firstItem.id, secondItem.id);
+        this.alertsService.showSuccessAlert('Barter offer accepted!');
+      });
+  }
+
+  private declineOtherTransactions(firstItemId: string, secondItemId: string): void {
+    this.elements.forEach(element => {
+        if (element.ownerItem.id === firstItemId || element.offeredItem.id === firstItemId ||
+          element.ownerItem.id === secondItemId || element.offeredItem.id === secondItemId) {
+          element.isCompleted = true;
```

Comment:
- [ ] can we detect what transaction was approved and what was removed/declined? if no - use status for transaction

Reply:
done